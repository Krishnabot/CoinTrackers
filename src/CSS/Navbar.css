#navbar {
  display: flex;
  justify-content: space-between;
  align-items: center;
  width: 90%;
  padding: 1rem 0.5rem;
}

.app-name {
  font-size: 1.2rem;
  color: #fff;
  font-weight: 600;
}

#nav-links {
  display: flex;
  list-style: none;
  gap: 1rem;
}

#nav-links li a {
  text-decoration: none;
  color: white;
}

#app-logo {
  background-image: url('../assets/image/coin_logo.png');
  background-size: cover;
  width: 50px;
  background-repeat: no-repeat;
  background-position: center;
}

.brand {
  display: flex;
}

@media all and (min-width: 768px) {
  .app-name {
    font-size: 1.5rem;
  }

  #nav-links li a {
    font-size: 1.2rem;
  }
}
