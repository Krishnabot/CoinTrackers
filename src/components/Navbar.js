import React from 'react';
import { NavLink, useLocation } from 'react-router-dom';
import '../CSS/Navbar.css';

const Navbar = () => {
  const location = useLocation();

  return (
    <nav id="navbar">
      <div className="brand">
        <div id="app-logo" />
        <h1 className="app-name">CoinTracker</h1>
      </div>

      <ul id="nav-links">
        <li>
          <NavLink to="/" id="home">
            {location.pathname === '/' ? 'Coins' : 'Back'}
          </NavLink>
        </li>
      </ul>
    </nav>
  );
};

export default Navbar;
