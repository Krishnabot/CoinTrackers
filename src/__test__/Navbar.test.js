import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import store from '../redux/configureStore';
import Navbar from '../components/Navbar';

describe('Test for nnav bar Rendering ', () => {
  test('Whether it matches match snapshot or not', () => {
    const tree = render(
      <BrowserRouter>
        <Provider store={store}>
          <Navbar />
        </Provider>
      </BrowserRouter>,
    );
    expect(tree).toMatchSnapshot();
  });
});
