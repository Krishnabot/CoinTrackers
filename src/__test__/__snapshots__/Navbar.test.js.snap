// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar rendering test match snapshot for Navbar 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <nav
        id="navbar"
      >
        <div
          class="brand"
        >
          <div
            id="app-logo"
          />
          <h1
            class="app-name"
          >
            CoinTracker
          </h1>
        </div>
        <ul
          id="nav-links"
        >
          <li>
            <a
              aria-current="page"
              class="active"
              href="/"
              id="home"
            >
              Coins
            </a>
          </li>
        </ul>
      </nav>
    </div>
  </body>,
  "container": <div>
    <nav
      id="navbar"
    >
      <div
        class="brand"
      >
        <div
          id="app-logo"
        />
        <h1
          class="app-name"
        >
          CoinTracker
        </h1>
      </div>
      <ul
        id="nav-links"
      >
        <li>
          <a
            aria-current="page"
            class="active"
            href="/"
            id="home"
          >
            Coins
          </a>
        </li>
      </ul>
    </nav>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
